pkgs/stdenv/generic/setup.sh

```diff
@@ -771,9 +771,18 @@ substitute() {
 }
 
 substituteInPlace() {
-    local fileName="$1"
-    shift
-    substitute "$fileName" "$fileName" "$@"
+    local -a fileNames=()
+    for arg in "$@"; do
+        if [[ "$arg" = "--"* ]]; then
+            break
+        fi
+        fileNames+=("$arg")
+        shift
+    done
+
+    for file in "${fileNames[@]}"; do
+        substitute "$file" "$file" "$@"
+    done
```

Comment:
```suggestion
    for arg in "$@"; do
        if [[ "$arg" = "--"* ]]; then
            break
        fi
        substitute "$file" "$file" "$@"
        shift
    done
```
maybe, I am not sure

Reply:
I think you can even just write `for arg; do`, and the `in "$@"` is the default.

Reply:
$file isn't set  and if it was then $@ would still contain rest of the filenames